### PR TITLE
Add additional logging to service client

### PR DESCRIPTION
**Pull request by abhipsaMisra**
From #1625 
Our log statements do not require a check for IsEnabled since that is already checked in the first implementation layer of the Logging class.

This is PR is just a sample on how the rest of the service client is going to look like when we are done with log instrumentations.

### CODE REVIEW

**Review by azabbasi — APPROVED** [review]

File: iothub/service/src/AmqpServiceClient.cs [hunk 1]
```diff
@@ -159,12 +155,26 @@ public override Task<PurgeMessageQueueResult> PurgeMessageQueueAsync(string devi
...
+                {
+                    { HttpStatusCode.NotFound, responseMessage => Task.FromResult((Exception)new DeviceNotFoundException(deviceId)) }
+                };
 
-            return _httpClientHelper.DeleteAsync<PurgeMessageQueueResult>(GetPurgeMessageQueueAsyncUri(deviceId), errorMappingOverrides, null, cancellationToken);
+                return _httpClientHelper.DeleteAsync<PurgeMessageQueueResult>(GetPurgeMessageQueueAsyncUri(deviceId), errorMappingOverrides, null, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                Logging.Error(this, $"PurgeMessageQueueAsync threw an exception: {ex}", nameof(PurgeMessageQueueAsync));
```
Comment (barustum) [inline iothub/service/src/AmqpServiceClient.cs:171]: nit: maybe change PurgeMessageQueueAsync to {nameof(PurgeMessageQueueAsync)}

**Review by barustum — APPROVED** [review]

File: iothub/service/src/AmqpServiceClient.cs (hunk 1 above)
Comment (abhipsaMisra) [reply iothub/service/src/AmqpServiceClient.cs:171]: sure!

**Pull request merged**